### DISCUSSION TITLE
the num of MemoryAccess per second

### DISCUSSION THREAD

**Discussion by Alma593** [sarchlab/mgpusim#86] [Q&A]
Mr. Sun

I want to know how many times per second the GPU can access memory when running a benchmark, 

so I record how many readrequests was generated in the ReadReqBuilder method in protocol.go, and then divide it by the kernel time to get an average of 2 billion readreqs generated per second, is this the right figure?

**Comment by syifan** [comment]
I do not think this is a good solution. All the components in the memory hierarchy use the `ReadRequestBuilder`. The CU can use it to generate requests for the L1 cache; the L1 cache needs to generate new requests for L2 caches. 

A better solution is to use a tracer to trace the DRAMs, or just to add a counter in the DRAM code and dump it at the end of the simulation.

**Comment by Alma593** [comment]
If the cache hits,it doesn't trigger a read or write operation on the DRAM?
If i trace DRAM, i still can not get the right number of memory access

**Reply by syifan** [reply]
You are right. You still cannot get the right number of memory access. Just say the most intuitive situation. The core sends a request that misses in the cache. The cache would need to generate another request to read from the memory. I guess you would count those as two requests? 

Also, the memory system is way more complex than a hit-miss model. If you want to calculate the number of transactions that the DRAM serves, it is always the best to count at the DRAM component.
